surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+- Yes, debugger on the device (e.g. an SSH session with a debugger running)
+- No
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging Rust?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Fixing a misbehaving program
+- Investigating a crash dump
+- Learning how a well-behaving program works
+- Profiling
+- Understand a third-party library's behaviour
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger [`NEXT`](#why-are-you-uncomfortable-using-a-debugger)
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
+Type: select all that apply
+
+- Unfamiliar (just need to spend more time)
+- Reliability (don't expect debugger to work well enough)
+- Ease-of-use
+- Lack of documentation with Rust
+
+> *Justification:*
+>
+> Why aren't users comfortable with debuggers is useful.
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you regularly use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- Go
+- C#
+- Other
+
+> *Justification:*
+>
+> Debugging experiences vary widely by language - debugging a interpreted
+> language like Python is very different from a compiled language like C/C++ -
+> this will inform user expectations with Rust debuggers.
+
+### How does debugging in language X compare to Rust?
+Type: freeform text
+
+> *Justification:*
+>
+> What can Rust's debugging experience learn from other languages.
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
```

Comment:
I'm not sure this is a useful question since even if there is correlation between one answer and a poor overall experience (and there would be a huge number of confounding factors), that wouldn't indicate causation

Reply:
I think the question is useful but less in terms of establishing causation and more for us to understand the context and environmental limitations solutions we think of have to operate within. 

For example, a lot of IDEs/editors might use gdb/lldb as the debugger but many of them don't provide the user with a way to run arbitrary debugger commands. They completely control the user's access to the debugger via their UI. For us, this might mean that adding new debugger commands via their extensibility API isn't helpful to these users and we should pursue other solutions instead.

Reply:
I've kept this, as Wesley describes, the answer could be useful for the working group